python/examples/agents/bee.py

```diff
@@ -83,62 +80,57 @@ def create_agent() -> BeeAgent:
-    #     reader.write("Agent 🤖 : ", "starting new iteration")
-    # elif event_meta.name == "success":
-    #     reader.write("Agent 🤖 : ", "success")
+    elif event.name == "update":
+        reader.write(f"Agent({data['update']['key']}) 🤖 : ", data["update"]["parsedValue"])
+    elif event.name == "start":
+        reader.write("Agent 🤖 : ", "starting new iteration")
+    elif event.name == "success":
+        reader.write("Agent 🤖 : ", "success")
 
 
 def observer(emitter: Emitter) -> None:
-    emitter.on("*.*", process_agent_events, EmitterOptions(match_nested=True))
+    emitter.on("*", process_agent_events, EmitterOptions(match_nested=False))
 
 
 async def main() -> None:
     """Main application loop"""
 
-    try:
-        # Create agent
-        agent = create_agent()
-
-        # Log code interpreter status if configured
-        code_interpreter_url = get_env_var("CODE_INTERPRETER_URL")
-        if code_interpreter_url:
-            reader.write(
-                "🛠️ System: ",
-                f"The code interpreter tool is enabled. Please ensure that it is running on {code_interpreter_url}",
-            )
-
-        reader.write("🛠️ System: ", "Agent initialized with LangChain Wikipedia tool.")
-
-        # Main interaction loop with user input
-        for prompt in reader:
-            # Run agent with the prompt
-            response = await agent.run(
-                BeeRunInput(prompt=prompt),
-                {
-                    "execution": {
-                        "max_retries_per_step": 3,
-                        "total_max_retries": 10,
-                        "max_iterations": 20,
-                    }
-                },
-            ).observe(observer)
-
-            reader.write("Agent 🤖 : ", response.result.text)
-
-    except Exception as e:
-        logger.error(f"Application error: {e!s}")
+    # Create agent
+    agent = create_agent()
+
+    # Log code interpreter status if configured
+    code_interpreter_url = os.getenv("CODE_INTERPRETER_URL")
+    if code_interpreter_url:
+        reader.write(
+            "🛠️ System: ",
+            f"The code interpreter tool is enabled. Please ensure that it is running on {code_interpreter_url}",
+        )
+
+    reader.write("🛠️ System: ", "Agent initialized with LangChain Wikipedia tool.")
+
+    # Main interaction loop with user input
+    for prompt in reader:
+        # Run agent with the prompt
+        response = await agent.run(
+            BeeRunInput(prompt=prompt),
+            BeeRunOptions(
+                execution=BeeAgentExecutionConfig(max_retries_per_step=3, total_max_retries=10, max_iterations=20)
+            ),
+        ).observe(observer)
+
+        reader.write("Agent 🤖 : ", response.result.text)
 
 
 if __name__ == "__main__":
-    # Run the async main function
-    asyncio.run(main())
+    try:
+        asyncio.run(main())
+    except FrameworkError as e:
+        print(e.explain())
+        sys.exit(1)
```

Comment:
As mentioned in my other comment, this update hides the error messaging when run by `test_example_execution` from #420, this is a comparison of two example with the same failing:
```
FAILED tests/examples/test_examples.py::test_example_execution[agents/bee.py] - SystemExit: 1
FAILED tests/examples/test_examples.py::test_example_execution[basic.py] - beeai_framework.backend.errors.ChatModelError: Stream response is empty.
```

With the context of #420 I'm unsure if this is the best way to handle this error